src/left.jl

```diff
@@ -11,32 +11,35 @@
 import LinearAlgebra: AdjointAbsVec, TransposeAbsVec
 
 # x = y'*A ⇐⇒ x' = (A'*y)
-Base.:(*)(y::AdjointAbsVec, A::LinearMap) = adjoint(*(A', y'))
+Base.:(*)(y::AdjointAbsVec, A::LinearMap) = adjoint(A' * y')
 Base.:(*)(y::TransposeAbsVec, A::LinearMap) = transpose(transpose(A) * transpose(y))
 
-# mul!(x, y', A)
-Base.@propagate_inbounds function LinearAlgebra.mul!(x::AbstractMatrix, y::AdjointAbsVec, A::LinearMap)
-    @boundscheck check_dim_mul(x, y, A)
-    @inbounds mul!(adjoint(x), A', y')
-    return adjoint(x)
-end
-
-Base.@propagate_inbounds function LinearAlgebra.mul!(x::AbstractMatrix, y::AdjointAbsVec,
-        A::LinearMap, α::Number, β::Number)
-    @boundscheck check_dim_mul(x, y, A)
-    @inbounds mul!(adjoint(x), A', y', α, β)
-    return adjoint(x)
-end
+# multiplication with vector/matrix
+for Atype in (AdjointAbsVec, Adjoint{<:Any,<:AbstractMatrix})
+    @eval Base.@propagate_inbounds function LinearAlgebra.mul!(x::AbstractMatrix, y::$Atype, A::LinearMap)
+        @boundscheck check_dim_mul(x, y, A)
+        @inbounds mul!(adjoint(x), A', y')
+        return x
```

Comment:
I think returning `adjoint(x)` was actually wrong, so that `mul!` returned a "standing vector". The way Julia's `mul!` writes into the `adjoint` wrapper of `x` is so beautiful.